### PR TITLE
Create specific status for Log & Span Decoration

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Store specific values in each Status depending ont he probe type

# Motivation
https://github.com/DataDog/dd-trace-java/pull/5048#discussion_r1163335694
# Additional Notes

### CODE REVIEW

File: dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/Snapshot.java [hunk 1]
```diff
@@ -636,7 +629,10 @@ private static boolean resolveEvaluateAt(
     public Status getStatus(String probeId) {
       Status result = statusByProbeId.get(probeId);
       if (result == null) {
-        return defaultStatus;
+        result = statusByProbeId.get(ProbeImplementation.UNKNOWN.getId());
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/Snapshot.java:632]: why do we need this?

File: dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/Snapshot.java [hunk 2]
```diff
@@ -672,85 +668,35 @@ public String toString() {
...
       public void addError(EvaluationError evaluationError) {
         errors.add(evaluationError);
       }
 
-      public void addTag(String tagName, String tagValue) {
-        if (tagsToDecorate == null) {
-          tagsToDecorate = new ArrayList<>();
-        }
-        tagsToDecorate.add(Pair.of(tagName, tagValue));
+      public boolean needFreeze() {
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/Snapshot.java:694]: Can we rename this method to make it more readable and easier to understand what this means.
I would suggest shouldFreezeContext or shouldCaptureContext.

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/ProbeDefinition.java [hunk 3]
```diff
@@ -163,6 +163,11 @@ public boolean hasCondition() {
     return false;
   }
 
+  @Override
+  public Snapshot.CapturedContext.Status createStatus() {
+    return null;
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/ProbeDefinition.java:168]: ins't this break the null pointer assumption? 

`statusByProbeId.computeIfAbsent(probeId, key -> probeImplementation.createStatus());`

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java [hunk 4]
```diff
@@ -387,21 +391,21 @@ public void commit(
...
-    Snapshot.CapturedContext.Status entryStatus = entryContext.getStatus(id);
-    Snapshot.CapturedContext.Status exitStatus = exitContext.getStatus(id);
+    LogStatus entryLogStatus = convertStatus(entryContext.getStatus(id));
+    LogStatus exitLogStatus = convertStatus(exitContext.getStatus(id));
     String message = null;
     switch (evaluateAt) {
       case ENTRY:
       case DEFAULT:
-        message = entryStatus.getMessage();
+        message = entryLogStatus.getMessage();
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java:400]: nit - do we need to rename?

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/ProbeDefinition.java (hunk 3 above)
Comment (shatzi) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/ProbeDefinition.java:168]: Maybe we should remove this @ Override as it implemented correctly at ProbeImplementation

File: dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/Snapshot.java [hunk 5]
```diff
@@ -672,85 +668,35 @@ public String toString() {
...
-      final ProbeImplementation probeImplementation;
-      boolean condition;
-      boolean hasLogTemplateErrors;
-      boolean hasConditionErrors;
-      String message;
-      // Span decoration
-      List<Pair<String, String>> tagsToDecorate;
+          new Status(ProbeImplementation.UNKNOWN) {
+            @Override
+            public boolean isCapturing() {
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/Snapshot.java:675]: Nit - I hope we can align the terms isCapturing and PASSING.

maybe rename to EMPTY_CAPTURING_STATUS (I don't like this) or change isCapturing to isActive / isEnabled or on the reverse isCanceled (I don't like that as well).

**Review by shatzi — APPROVED** [review]

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/ProbeDefinition.java (hunk 3 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/ProbeDefinition.java:168]: > Maybe we should remove this @ Override as it implemented correctly at ProbeImplementation

I dont get it! can you elaborate?

File: dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/Snapshot.java (hunk 2 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/Snapshot.java]: done

File: dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/Snapshot.java (hunk 5 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/Snapshot.java]: I prefer capturing then

**Pull request merged**